rasa/core/actions/forms.py

```diff
@@ -523,8 +528,11 @@ async def _validate_if_required(
            - the form is called after `action_listen`
            - form validation was not cancelled
         """
-        # no active_loop means that it is called during activation
-        needs_validation = not tracker.active_loop or (
+        # No active_loop means there are no form filled slots to validate yet
+        if not tracker.active_loop:
+            return []
+
+        needs_validation = (
```

Comment:
@tayfun I have a retrospective question about this change.
I _think_ it's the thing that is causing the following behaviour between 2.x and 3.x to change: In 2.x, it was possible to add "form entry" logic to `FormValidationAction` by doing a check on whether `requested_slot` was `None`. Now that the form validation action is guaranteed not to run until the loop is active, there's nowhere to "hook into" the form before it starts.
Do you know if this was a conscious decision or is it an unintended side-effect?

Reply:
I think it was a side effect, though I'm not 100% sure about slots feature in general. This does feel like a hack for "form entry" logic though, I wonder if there is another way so that form validation action can distinguish form entry or if we should add a new explicit way. @ancalita thoughts?

Reply:
Oh this was to remove the code between L361-363 further up  in original `forms.py` + I also think it's confusing to allow a `FormValidationAction` to run when the form is not active yet / just being activated

Reply:
I remember adding that L361 code to prevent some unit tests from breaking.

Reply:
It's true that there's no slots to validate yet, but since a `FormValidationAction` is there to validate form input in general, it does make sense that it could also "prepare the way" for form input when it has just been activated. What was the specific unwanted effect of letting the validation action run here? Maybe there's a way to mitigate that without completely disabling the form validation from running when it's just starting.

Reply:
In general though I would advocate having a separate method like `run_on_activation` instead of having logic added directly to `run` with checks on `requested_slot` to see if it's the start of a form. That would need to be a bit more thought out though since it would be new behaviour.

Reply:
Request opened [here](https://github.com/RasaHQ/rasa/issues/10913)